.gemini/styleguide.md

```diff
@@ -0,0 +1,293 @@
+# qsim style guide
+
+This style guide outlines the coding conventions for this project. There are
+separate subsections for Python, C++, and other file types below.
+
+## General guidance
+
+### Overall principles
+
+*   _Readability_: Code should be easy to understand for all team members.
+
+*   _Maintainability_: Code should be easy to modify and extend.
+
+*   _Consistency_: Adhering to a consistent style across all projects improves
+    collaboration and reduces errors.
+
+*   _Performance_: While readability is paramount, code should be efficient.
+
+### Overall development approach
+
+*   Use test-driven development (TDD) whenever possible. Write a failing test
+    _before_ writing new code. This ensures testability and clarifies
+    requirements.
+
+*   Isolate tests. Tests must be independent and must not rely on the state of
+    other tests. Use setup and teardown functions to create a clean environment
+    for each test run.
+
+*   When new functions, classes, and files are introduced, they should also have
+    corresponding tests. Existing tests must continue to pass (or be updated)
+    when changes are introduced, and code should be covered by tests.
+
+*   Make sure to cover edge cases: Test for invalid inputs, null values, empty
+    arrays, zero values, and off-by-one errors.
+
+*   Use mocking for dependencies. In unit tests, external dependencies (e.g.,
```

Comment:
The only external dependency of the C++ code is Eigen3. This dependence is not propagated to qsimcirq.